eZ/Publish/Core/REST/Server/Controller/Location.php

```diff
@@ -69,14 +83,20 @@ public function __construct(LocationService $locationService, ContentService $co
      */
     public function redirectLocation(Request $request)
     {
-        if (!$request->query->has('id') && !$request->query->has('remoteId')) {
-            throw new BadRequestException("At least one of 'id' or 'remoteId' parameters is required.");
+        if (!$request->query->has('id') &&
+            !$request->query->has('remoteId') &&
+            !$request->query->has('urlAlias')
+        ) {
+            throw new BadRequestException("At least one of 'id', 'remoteId' or 'urlAlias' parameters is required.");
         }
 
         if ($request->query->has('id')) {
             $location = $this->locationService->loadLocation($request->query->get('id'));
-        } else {
+        } elseif ($request->query->has('remoteId')) {
             $location = $this->locationService->loadLocationByRemoteId($request->query->get('remoteId'));
+        } else {
```

Comment:
Hmm, code to be changed is relatively small so this seems like a good chance for some optimization/refactoring.

Instead of initial check that leads to `BadRequestException` please add here an extra `else if ($request->query->has('urlAlias'))` and throw in the last `else`.

This would allow us to:
1) Reduce number of calls to `$request->query->has`
2) Make if-else blocks look more readable.